src/parse_impl.rs

```diff
@@ -0,0 +1,167 @@
+use crate::parse_fn::try_consume_fn;
+use crate::parse_utils::{consume_attributes, consume_stuff_until, consume_vis_marker};
+use crate::types::{Constant, ImplMember, TyDefinition, ValueExpr};
+use crate::types_edition::GroupSpan;
+use crate::{Attribute, TyExpr, VisMarker};
+use proc_macro2::{Group, Ident, TokenTree};
+use std::iter::Peekable;
+
+type TokenIter = Peekable<proc_macro2::token_stream::IntoIter>;
+
+pub(crate) fn consume_for(tokens: &mut TokenIter) -> Option<Ident> {
+    if let Some(TokenTree::Ident(ident)) = tokens.peek() {
+        if ident == "for" {
+            let tk_for = ident.clone();
+            tokens.next();
+            return Some(tk_for);
+        }
+    }
+
+    None
+}
+
+pub(crate) fn consume_constant(
+    tokens: &mut TokenIter,
+    attributes: Vec<Attribute>,
+    vis_marker: Option<VisMarker>,
+) -> Constant {
+    let tk_const = match tokens.next() {
+        Some(TokenTree::Ident(ident)) if ident == "const" => ident,
+        _ => panic!("cannot parse constant"),
+    };
+
+    let name = match tokens.next() {
+        Some(TokenTree::Ident(ident)) => ident,
+        _ => panic!("cannot parse constant"),
+    };
+
+    let tk_colon = match tokens.next() {
+        Some(TokenTree::Punct(punct)) if punct.as_char() == ':' => punct,
+        _ => panic!("cannot parse constant"),
+    };
+
+    let ty_tokens = consume_stuff_until(
+        tokens,
+        |tt| matches!(tt, TokenTree::Punct(punct) if punct.as_char() == '='),
+        true,
+    );
+
+    let tk_equals = match tokens.next() {
+        Some(TokenTree::Punct(punct)) if punct.as_char() == '=' => punct,
+        _ => panic!("cannot parse constant"),
+    };
+
+    let value_tokens = consume_stuff_until(
+        tokens,
+        |tt| matches!(tt, TokenTree::Punct(punct) if punct.as_char() == ';'),
+        true,
+    );
```

Comment:
Constants without `= xxx` are syntactically valid and should be accepted.

Reply:
Implemented, the `=` and value are now optional:
```rs
pub struct Constant {
    pub attributes: Vec<Attribute>,
    pub vis_marker: Option<VisMarker>,
    pub tk_const: Ident,
    pub name: Ident,
    pub tk_colon: Punct,
    pub ty: TyExpr,
    /// The '=' sign is optional; constants without initializer are syntactically valid.
    pub tk_equals: Option<Punct>,
    /// The initializer value is optional; constants without initializer are syntactically valid.
    pub initializer: Option<ValueExpr>,
    pub tk_semicolon: Punct,
}
```